Optimize Docker and WDL

nolwarre opened the pull request:
Optimized Dockerfile to only create necessary files and organize to be more readable

Optimized WDL to get rid of long paths and generate sampleID from contigFasta

nolwarre committed:
Remove excess testing setup

nolwarre committed:
simplify paths

nolwarre committed:
modify handeling of output name, got rid of sampleID parameter

nolwarre committed:
Move set -eux -o pipefail to top of command
localize input to working dir

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
+    assembledMitoGFF=(./mitogenome.annotation/mitogenome.annotation_MitoFinder_mitfi_Final_Results/mitogenome.annotation_mtDNA_contig.gff)
+    assembledMitoFasta=(./mitogenome.annotation/mitogenome.annotation_MitoFinder_mitfi_Final_Results/mitogenome.annotation_mtDNA_contig.fasta)
 
     # finds the number of bases to rotate the mitogenome to correctly align
```
You might want to be more explicit about what you are grepping for here.

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
+      -i $assembledMitoFasta \
+      -r $numRotation > ~{mitoOut}
   >>>
   #specify the output(s) of this task so cromwell will keep track of them
```
I'm not sure you need this comment.

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
+      -r $numRotation > ~{mitoOut}
   >>>
   #specify the output(s) of this task so cromwell will keep track of them
   output {
```
Please add `preemptible: 1` to your runtime section. It's not the biggest deal here, but it's good practice for cost savings.

juklucas commented on MitoHiFi.wdl:
```diff
@@ -8,7 +8,7 @@ version 1.0
 #add and name a workflow block
```
I think you can drop this comment.

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
   #define command to execute when this task runs
   String mitoOut = basename(contigsFasta,".fa") + ".chrM.fa"
```
This call to basename only handles files with *.fa, not *.fasta, and it doesn't handle gzipped files (which is what we typically store.)

We have used the ridiculously ugly:
`$(basename ~{assemblyFasta} | sed 's/.gz$//' | sed 's/.fa\(sta\)*$//' | sed 's/[._][pm]at\(ernal\)*$//')`
In some of our other workflows.

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
-    p=$(pwd)
-    # go to work directory
-    cd /opt/MitoHiFi/exampleFiles/
+    # Set the exit code of a pipeline to that of the rightmost command
```
It is also ok to write this as
`set -eux -o pipefail`

juklucas commented on MitoHiFi.wdl:
```diff
@@ -27,47 +26,45 @@ task mito {
+    # localize fasta input to working directory
+    mv ~{contigsFasta} localContigs
 
     # run main MitoHiFi using parameters
```
This comment doesn't actually tell the reader anything.

Maybe something like: Re-assemble mito contig from input (raw) assembly.

juklucas commented on Dockerfile:
```diff
@@ -41,9 +41,6 @@ ENV PATH=/opt/ncbi-blast-2.11.0+/bin:$PATH
 WORKDIR /opt
 RUN git clone https://github.com/marcelauliano/MitoHiFi.git && \
     cd MitoHiFi && \
```
When installing from GitHub (but not from a specific version) it might be a good idea to pin to a specific commit. It looks like this: `git reset --hard 1926400fcb098ab3e4745ea0872d23f91d730340`

Check here (https://github.com/human-pangenomics/hpp_production_workflows/blob/master/QC/docker/merqury/Dockerfile) for an example.

In order to find the commit, just go to the GitHub repo, and click on the commits area. Then you can find a suitable commit (the most recent one, in this case).

This helps with reproducibility, and I guess to some extent protects you from changes in their code in the off-chance that you need to rebuild your Docker image.

nolwarre replied:
Added this for both MitoHiFi and Mitofinder now

juklucas reviewed (commented):
This looks great. All the comments I gave are window dressing (with the exception of handling gzipped assembly files).

Also, you should think about how to ensure that the result isn't the reverse complement of what we want! If you do want to flip, then you can call samtools faidx. It is testing whether or not you need to flip which might be hard. Maybe the answer is in the gff file?

nolwarre committed:
Explict comments
added preemptible parameter
robust sample name handling
reproducibility in docker container

nolwarre committed:
added dockstore.yml

nolwarre committed:
Modified .dockstore.yml

nolwarre committed:
Added unzip for .gz

nolwarre committed:
Added --force to force rotations

nolwarre committed:
Added --force to force rotations

nolwarre committed:
Merge branch 'optimize-docker-wdl' of github.com:nolwarre/MitoHiFi-Workflow into optimize-docker-wdl

nolwarre committed:
forget new line break
